pkg/newlog/cmd/newlogd.go

```diff
@@ -1019,18 +1020,22 @@ func doMoveCompressFile(tmplogfileInfo fileChanInfo) {
 	outfile := gzipFileNameGet(isApp, timenowNum, dirName, appuuid, tmplogfileInfo.notUpload)
 
 	// read input file
-	ifile, err := os.Open(tmplogfileInfo.tmpfile)
+	// with double overhead for limit
+	content, err := utils.ReadWithMaxSize(log, tmplogfileInfo.tmpfile, int(2*maxLogFileSize))
```

Comment:
Can't we do better by creating a reader and feeding the reader into a gzip function operating on a reader?